prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -3831,15 +3831,15 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             // TODO: Mmm... are these parameters correct?
                             let encoded_spec = self.encoder.encode_assertion(
                                 &assertion,
-                                self.mir,
+                                &self.mir,
                                 PRECONDITION_LABEL,
                                 &encoded_args,
                                 None,
                                 false,
                                 Some(loop_inv_block),
                                 ErrorCtxt::GenericExpression,
                             );
-                            let spec_spans = typed::Spanned::get_spans(assertion, self.encoder.env().tcx());
+                            let spec_spans = typed::Spanned::get_spans(assertion, &self.mir, self.encoder.env().tcx());
```

Comment:
If this is a bugfix, is it easy to add a test to avoid future regressions?

Reply:
I don't think span positions are tested. The change to `get_spans` may just slightly improve error reporting?

Reply:
In that case, we should see a difference in a "ui" test.

Reply:
We should, although the current test suite does not seem to have such a case. As you can see, the CI passes. To be honest, I am not completely sure how to cause an error that would be reported on the whole `ForAll` assertion, rather than its body.

Reply:
I see, I thought you did this to fix something in the output. As far as I know we always report failing `ForAll` pointing to their body, just because it's easy to reuse and merge together the spans that come from the compiler.